<!--  -->
<launch>
  <node name="joy" pkg="joy" type="joy_node" output="screen">
    <param name="dev" type="string" value="/dev/input/js0"/>
    <param name="deadzone" value="0.0"/>
    <param name="autorepeat_rate" value="0"/>
  </node>

  <node pkg="mrop" type="simulator" name="simulator" output="screen">
    <param name="parent_frame"       value="world"/>
    <param name="child_frame"        value="virtual"/>
    
    <param name="active"             value="true"/>

    <param name="loop_rate"          value="100.0"/>    
    <param name="time_constant"      value="0.0"/>
    <param name="time_delay"         value="0.0"/>
  </node>

  <node pkg="mrop" type="manual_controller" name="manual_controller" output="screen">
    <param name="active"        value="true"/>
    <param name="use_keys"      value="true"/>
    <param name="use_joy"       value="true"/>

    <param name="linear_gain"   value="0.25"/>
    <param name="angular_gain"  value="1.0"/>
  </node>

  <node pkg="mrop" type="reference_generator" name="reference_generator" output="screen">
    <param name="parent_frame"           value="world"/>
    <param name="child_frame"            value="reference"/>
    
    <param name="active"                 value="true"/>
    <param name="trajectory_paused"      value="false"/>
    <param name="trajectory_stopped"     value="false"/>

    <param name="loop_rate"              value="100.0"/>    

    <param name="trajectory_type"        value="2"/>

    <param name="initial_x"              value="0.0"/>
    <param name="initial_y"              value="0.0"/>
    <param name="initial_theta"          value="0.0"/>
    <param name="linear_velocity"        value="0.1"/>
    <param name="harmonic_period"        value="5.0"/>
    <param name="harmonic_radius_x"      value="1.0"/>
    <param name="harmonic_radius_y"      value="1.0"/>
    <param name="harmonic_multiplier_x"  value="1.0"/>
    <param name="harmonic_multiplier_y"  value="1.0"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz"/>
</launch>

